pkg/util/validation/validate.go

```diff
@@ -93,10 +94,12 @@ func ValidateLabels(cfg LabelValidationConfig, userID string, ls []client.LabelA
 	metricName, err := extract.MetricNameFromLabelAdapters(ls)
 	if cfg.EnforceMetricName(userID) {
 		if err != nil {
+			DiscardedSamples.WithLabelValues(invalidMetricName, userID).Inc()
```

Comment:
This case isn't an invalid metrics name, but rather a missing metric name.  Can you make a new reason string for this?

Reply:
Thanks for adding it Tom.